### PR TITLE
O365 Opening Hours Synchronization

### PR DESCRIPTION
**Pull request by johlindq**
## Supports syncing opening hours from Outlook Calendar.

Opening hours for resources that are linked to an outlook calendar are
managed by creating meetings called "Varattavissa Varaamo" in Outlook
calendar.

### CODE REVIEW

**Review by aceViilee — APPROVED** [review]
Consider moving keystring "Varattavissa Varaamo" to easily configurable location. For instance through django admin panel. Also key strings should be constants

File: respa_o365/respa_availabilility_repository.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        return {r.id: (ChangeType.NO_CHANGE, period_change_key(r)) for r in periods}, new_memento
+
+    def get_changes_by_ids(self, item_ids, memento=None):
+        periods = Period.objects.filter(id__in=item_ids)
+        new_memento = datetime(1970, 1, 1, tzinfo=timezone.utc).strftime(time_format)
+        return {r.id: (ChangeType.NO_CHANGE, period_change_key(r)) for r in periods}, new_memento
+
+
+
+    def get_changes(self, memento=None):
```
Comment (SanttuA) [inline respa_o365/respa_availabilility_repository.py:62]: Same function defined multiple times

File: respa_o365/respa_availabilility_repository.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    if reservation.modified_at <= time:
+        return ChangeType.NO_CHANGE
+    if reservation.created_at > time:
+        return ChangeType.CREATED
+    return ChangeType.UPDATED
+
+def period_change_key(item):
+    # Changing periods through other means is prevented when there are calendar links. Thus the periods do not change
+    # on the Respa side and it is fine to always return the same change key.
+    return ""
```
Comment (SanttuA) [inline respa_o365/respa_availabilility_repository.py:91]: If the goal is to always use the same key regardless of item/period, wouldn't a variable make more sense? The key value could be something better than an empty string.

**Comment by johlindq** [comment]
> Consider moving keystring "Varattavissa Varaamo" to easily configurable location. For instance through django admin panel. Also key strings should be constants

@aceViilee Would it be OK to make it configurable through the .env file? I don't think it is something we will be changing a lot.

**Comment by aceViilee** [comment]
> 
> 
> > Consider moving keystring "Varattavissa Varaamo" to easily configurable location. For instance through django admin panel. Also key strings should be constants
> 
> @aceViilee Would it be OK to make it configurable through the .env file? I don't think it is something we will be changing a lot.

.env file is ok. Logic can be rewritten at some point, but as of now ok

**Review by aceViilee — APPROVED** [review]

**Review by ezkat — APPROVED** [review]

**Review by SanttuA — APPROVED** [review]

**Pull request merged**